### PR TITLE
Price/Fee display UI change - Swap page and Confirmation Modal

### PR DESCRIPTION
**Pull request by W3stside**
# Summary

Blocked by #1086 

Closes the final price display changes in the wireframes here: https://www.figma.com/file/bcoZf5RIgVKDItdOaTZq18/GP-Swap?node-id=1310%3A2

SELL ORDER
image

BUY ORDER
image

FEE > INPUT
image

  # To Test

1. make sell order
2. make buy order
3. input tiny amt of weth to see fee error also shows up

### CODE REVIEW

File: src/custom/pages/Swap/index.tsx [hunk 1]
```diff
@@ -27,6 +27,11 @@ import { LONG_LOAD_THRESHOLD, SHORT_PRECISION } from 'constants/index'
 import { formatSmart } from 'utils/format'
 import { MouseoverTooltipContent } from 'components/Tooltip'
 import { StyledInfo } from 'pages/Swap/SwapMod'
+import { Repeat } from 'react-feather'
+import { Trans } from '@lingui/macro'
+import TradePrice from 'components/swap/TradePrice'
+import TradeGp from 'custom/state/swap/TradeGp'
```
Comment (W3stside) [inline src/custom/pages/Swap/index.tsx:33]:
```suggestion
import TradeGp from 'state/swap/TradeGp'
```

File: src/custom/pages/Swap/index.tsx [hunk 2]
```diff
@@ -163,7 +219,8 @@ function FeeGreaterMessage({ fee }: FeeGreaterMessageProp) {
         </MouseoverTooltipContent>
       </RowFixed>
       <TYPE.black fontSize={14} color={theme.text1}>
-        {formatSmart(fee, SHORT_PRECISION)} {fee.currency.symbol}
+        {formatSmart(fee, SHORT_PRECISION)} {fee.currency.symbol}{' '}
```
Comment (W3stside) [inline src/custom/pages/Swap/index.tsx:222]: PRECISION = 4 here. up for discussion. looks better smaller decimals as there's already lots of data in this line

**Review by anxolin — COMMENTED** [review]
Should these two be coherent? (label & Value)

image

image

**Comment by W3stside** [comment]
> Should these two be coherent? (label & Value)
> 
> image
> 
> image

nice catch. done in e071463

File: src/custom/pages/Swap/index.tsx (hunk 2 above)
Comment (alfetopito) [reply src/custom/pages/Swap/index.tsx]: For the fee I think it's best to use the same as other amounts.
What I agree that should be changed is the USD estimations and percentages, to use only 2 decimals.
But that's not in the scope of this PR

File: src/custom/components/swap/TradePrice/index.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import React, { useMemo } from 'react'
+import TradePriceMod, { TradePriceProps } from './TradePriceMod'
+import { useUSDCValue } from 'hooks/useUSDCPrice'
+import { formatSmart } from 'utils/format'
+import { tryParseAmount } from 'state/swap/hooks'
+import { SHORT_PRECISION } from '@src/custom/constants'
```
Comment (alfetopito) [inline src/custom/components/swap/TradePrice/index.tsx:6]: ?

```suggestion
import { SHORT_PRECISION } from 'constants/index'
```

File: src/custom/components/swap/TradePrice/index.tsx [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+  const priceSide = useMemo(
+    () =>
+      !showInverted
+        ? tryParseAmount(price.invert().toFixed(), price.baseCurrency)
+        : tryParseAmount(price.toFixed(), price.quoteCurrency),
+    [price, showInverted]
+  )
+  const amount = useUSDCValue(priceSide)
+  const fiatValueFormatted = formatSmart(amount, SHORT_PRECISION)
```
Comment (alfetopito) [inline src/custom/components/swap/TradePrice/index.tsx:21]: SHORT_PRECISION === 4?
(See my other comment) Can we use 2 for fiat?

File: src/custom/pages/Swap/index.tsx [hunk 5]
```diff
@@ -142,17 +149,72 @@ export interface SwapProps extends RouteComponentProps {
...
+
+function FeeGreaterMessage({ trade, fee }: FeeGreaterMessageProp) {
   const theme = useContext(ThemeContext)
+  // trades are null when there is a fee quote error e.g
+  // so we can take both
+  const feeAmount = trade?.fee.feeAsCurrency || fee
+  const feeFiatValue = useUSDCValue(feeAmount)
+
+  const { realizedFee } = computeTradePriceBreakdown(trade)
+  const feeFiatDisplay = `(≈$${formatSmart(feeFiatValue, SHORT_PRECISION)})`
```
Comment (alfetopito) [inline src/custom/pages/Swap/index.tsx:211]: 2 decimals :)

**Review by alfetopito — COMMENTED** [review]
Minor comments, probably to be addressed in a different PR and one potential issue:

Screenshot from 2021-07-27 09-34-19

Fee USD is not available and the amount in the tooltip is `undefined`.
Also, should we have that tooltip?

File: src/custom/components/swap/TradePrice/index.tsx (hunk 4 above)
Comment (W3stside) [reply src/custom/components/swap/TradePrice/index.tsx:21]: yes!

File: src/custom/pages/Swap/index.tsx (hunk 5 above)
Comment (W3stside) [reply src/custom/pages/Swap/index.tsx]: yes!

File: src/custom/components/swap/TradePrice/index.tsx (hunk 4 above)
Comment (W3stside) [reply src/custom/components/swap/TradePrice/index.tsx]: c200c53

File: src/custom/pages/Swap/index.tsx (hunk 5 above)
Comment (W3stside) [reply src/custom/pages/Swap/index.tsx]: c200c53

**Comment by W3stside** [comment]
> Minor comments, probably to be addressed in a different PR and one potential issue:
> 
> Screenshot from 2021-07-27 09-34-19
> 
> Fee USD is not available and the amount in the tooltip is `undefined`.
> Also, should we have that tooltip?

298741d i mean we can idk

**Comment by alfetopito** [comment]
To be sure I'm not seeing outdated stuff, I've loaded it locally.

I still see the `undefined` tooltip
screenshot_2021-07-27_14-41-14

If I switch the price over though, the price is shown no problem
screenshot_2021-07-27_14-44-02

So, 2 issues:
1. Title with `undefined` -- for this I'm in favour of removing the title
2. Only see an amount if I invert the price -- for this I don't see why we don't get a price :thinking: is 1 AMP too small?
screenshot_2021-07-27_14-47-15

~0.7 cents. Should still be enough to show something...

**Comment by W3stside** [comment]
> To be sure I'm not seeing outdated stuff, I've loaded it locally.
> 
> I still see the `undefined` tooltip
> screenshot_2021-07-27_14-41-14
> 
> If I switch the price over though, the price is shown no problem
> screenshot_2021-07-27_14-44-02
> 
> So, 2 issues:
> 
> 1. Title with `undefined` -- for this I'm in favour of removing the title
> 2. Only see an amount if I invert the price -- for this I don't see why we don't get a price 🤔 is 1 AMP too small?
>    screenshot_2021-07-27_14-47-15
> 
> ~0.7 cents. Should still be enough to show something...

in the console u can check the trade obj it's on the window. so: `trade.executionPrice.invert().toFixed(12)` = `0.000030308007` which in the scheme of our formatter is just 0

**Review by anxolin — COMMENTED** [review]
Looks great, only tiny detail: can we remove this USD estimation?

image

**Comment by elena-zh** [comment]
HI @W3stside , great job! 
Some minor issues: 
1. I'm not sure whether it is an issue or not, but 'Slippage tolerance' appears on the Swap form when I change it manyally in the settings. When the setting is 'Auto', 'Slippage tolerance' is not displayed on the form
https://drive.google.com/file/d/19wjFS41uX3qWb6h4cf870R_542KOxlkC/view

2. Price USD estimation is different on the Swap form and in the Confirmation modal (difference is about 0.01$). I assume, it should be the same
image


3. Prices are jumping when invert them: 
 - when USD amount is loaded: https://drive.google.com/file/d/1ecadUoMNmCDuz2lbQk_SquVfM4CVSnPN/view
 - when invert them on a Mobile screen: https://drive.google.com/file/d/1QogvmwHlS3pLzxkwMb17HsJFsPQj-LgW/view
 - when invert prices on the Confirmation modal on a Mobile screen: https://drive.google.com/file/d/1H4yIb5qfnAkCWkhd6yRPRO5h8Sj0yEbR/view

4. Not a priority, but Price/Fee/Slippage values sometimes look like have a different alignment.. IMHO
image

**Comment by W3stside** [comment]
> HI @W3stside , great job!
> Some minor issues:
> 
> 1. I'm not sure whether it is an issue or not, but 'Slippage tolerance' appears on the Swap form when I change it manyally in the settings. When the setting is 'Auto', 'Slippage tolerance' is not displayed on the form
>    https://drive.google.com/file/d/19wjFS41uX3qWb6h4cf870R_542KOxlkC/view
> 2. Price USD estimation is different on the Swap form and in the Confirmation modal (difference is about 0.01$). I assume, it should be the same
>    image
>    

☝️ likely a small price change in between, happens a lot
> 3. Prices are jumping when invert them:
> 
> * when USD amount is loaded: https://drive.google.com/file/d/1ecadUoMNmCDuz2lbQk_SquVfM4CVSnPN/view
> * when invert them on a Mobile screen: https://drive.google.com/file/d/1QogvmwHlS3pLzxkwMb17HsJFsPQj-LgW/view
> * when invert prices on the Confirmation modal on a Mobile screen: https://drive.google.com/file/d/1H4yIb5qfnAkCWkhd6yRPRO5h8Sj0yEbR/view
> 

☝️ agreed it's not beautiful but for now I would say low prio and we leave as is. solution could be to have a placeholder e.g '$...' or something but there would still be some jumping
> 1. Not a priority, but Price/Fee/Slippage values sometimes look like have a different alignment.. IMHO
>    image

☝️ agree this is ugly, tho i'm not skilled enough to fix it quickly. @biocom maybe add it to an issue for the future

**Comment by W3stside** [comment]
> Looks great, only tiny detail: can we remove this USD estimation?
> 
> image

#1086

**Comment by biocom** [comment]
Can address @W3stside 's point 3 about jumping when reversing: I think a good solution is to 'wrap' the USD estimate down on mobile to prevent this. Will address separately.

**Review by alfetopito — APPROVED** [review]
Looking good!

Only one small thing still not working that I fixed in another PR https://github.com/gnosis/cowswap/pull/1091

**Review by anxolin — APPROVED** [review]
Great!

The base tokens don't work for other networks other than mainnet. Although not from this PR.

**Pull request merged**